cmake/modules/BuildJaeger.cmake

```diff
@@ -0,0 +1,75 @@
+###############################################################################
+#Ceph - scalable distributed file system
+#
+#Copyright (C) 2020 Red Hat Inc.
+#
+#This is free software; you can redistribute it and/or
+#modify it under the terms of the GNU Lesser General Public
+#License version 2.1, as published by the Free Software
+#Foundation.  See file COPYING.
+################################################################################
+#
+# This module builds Jaeger after it's dependencies are installed and discovered
+# opentracing: is built using cmake/modules/Buildopentracing.cmake
+# Thrift: build using cmake/modules/Buildthrift.cmake
+# yaml-cpp, nlhomann-json: are installed locally and then discovered using
+# Find<package>.cmake
+# Boost Libraries used for building thrift are build and provided by
```

Comment:
do you mean the thrift does not compile with the boost libraries shipped by distro?

Reply:
yes, for some distro like ubuntu the older version of Boost does not support the latest thrift. ( we can update this once we drop support for bionic)

Reply:
boost 1.72 is required since 713efb66e8d7b2585dfd6783f8f23ce409b50871. so i guess it's new enough?

Reply:
@tchaikov I rebased and am trying to eliminate installing boost altogether as well. hitting this issue on jenkins for opensuse and focal:
```
build/ceph-16.0.0-6677-gea00988b/obj-x86_64-linux-gnu/external/include/thrift/transport/TTransportException.h:23:10: fatal error: boost/numeric/conversion/cast.hpp: No such file or directory
   23 | #include <boost/numeric/conversion/cast.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I will update once i can reproduce this and fix it.

Reply:
@tchaikov resolved this issue updated code to remove installing boost, it was issue with Jaeger not including Boost headers.
https://shaman.ceph.com/builds/ceph/wip-jaegertracing-in-ceph/